packages/@lwc/engine/src/libs/mutation-tracker/__tests__/index.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+declare var describe: any;
+declare var it: any;
+declare var expect: any;
+
+import { ReactiveObserver, valueMutated, valueObserved } from '../index';
+
+describe('reactive-service', () => {
+    it('should observe object mutations via ReactiveObserver() instance during the observing phase', () => {
+        let changed = false;
+        const o = { x: 1 };
+        // listening for mutation on object `o`
+        const ro = new ReactiveObserver(() => {
+            changed = true;
+        });
+        ro.observe(() => {
+            valueObserved(o, 'x');
+        });
+        // mutating object `o`
+        o.x = 2;
+        valueMutated(o, 'x');
+        expect(changed).toBe(true);
+        // resetting the observer
+        changed = false;
+        ro.reset();
+        // mutating object `o` again
+        o.x = 3;
+        valueMutated(o, 'x');
+        expect(changed).toBe(false);
+    });
+    it('should support observing and mutating the same value multiple times', () => {
```

Comment:
@caridy this (maybe other test?!) is throwing in the ci with error:
```
@lwc/engine: PASS lwc-engine src/libs/mutation-tracker/__tests__/index.spec.ts
@lwc/engine: (node:308) UnhandledPromiseRejectionWarning: Error: expect(received).toBe(expected) // Object.is equality
@lwc/engine: Expected: 2
@lwc/engine: Received: 1
@lwc/engine: (node:308) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
@lwc/engine: (node:308) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
does not seems to be doing anything async, or it does?

Reply:
this is a weird thing that I have been for a while... I don't know what is it, but it is not from this PR, I Can confirm that.